tests/unit/test_io.py

```diff
@@ -24,6 +24,10 @@ def test_serialization(
     io.write(exported, filename)
     asdict = io.asdict(exported)
     imported = io.fromdict(asdict)
+    io.write(
+        imported,
+        output_dir + f"test_write_read_{formalism}.imported.{file_extension}",
+    )
```

Comment:
That's a separate test for testing the write functionality? Maybe this can be put into a separate test. To keep the tests more distinct

Reply:
No, I put write statements in there so you can still view the output in case there's a problem. The writing itself doesn't really need to be tested, because that would essentially be testing the `json` or `yaml` package (or, whether the nested `dict` can be dumped to those formats).

Reply:
Ok actually that line was some outdated code, fixed by ec8c489